### PR TITLE
Chore/semaphore v3

### LINKED ISSUE #16: Upgrade Semaphore dependencies in the boilerplate

### PR DESCRIPTION
**Pull request by vplasencia**
## Description

Add script to copy contract artifacts to `subgraph` and `web-app` folders and info in the Readme file about the steps to follow every time the smart contract is deployed.

## Related Issue
Closes #16 

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
         "start": "yarn workspace web-app dev",
         "lint": "eslint . --ext .js,.ts",
         "prettier": "prettier -c .",
-        "prettier:write": "prettier -w ."
+        "prettier:write": "prettier -w .",
+        "copy:contract-artifacts": "npx ts-node scripts/copy-contract-artifacts.ts"
```
Comment (cedoor) [inline package.json:14]:
```suggestion
"copy:contract-artifacts": "ts-node scripts/copy-contract-artifacts.ts"
```
`npx` shouldn't be necessary.

File: scripts/copy-contract-artifacts.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import * as fs from "fs"
+
+const contractArtifactsPath = "apps/contracts/build/contracts/contracts/Feedback.sol"
+
+const subgraphArtifactsPath = "apps/subgraph/contract-artifacts"
+
+const webAppArtifactsPath = "apps/web-app/contract-artifacts"
+
+async function copyArtifactsFiles() {
```
Comment (cedoor) [inline scripts/copy-contract-artifacts.ts:9]:
```suggestion
async function main() {
   try {
        await fs.promises.copyFile(`${contractArtifactsPath}/Feedback.json`, `${subgraphArtifactsPath}/Feedback.json`)
        await fs.promises.copyFile(`${contractArtifactsPath}/Feedback.json`, `${webAppArtifactsPath}/Feedback.json`)
    } catch (err) {
        throw new Error("Error occurred while copying the contract artifacts files")
    }
}

main()
    .then(() => process.exit(0))
    .catch((error) => {
        console.error(error)
        process.exit(1)
    })
```

What about using the Hardhat standard we already used here? https://github.com/semaphore-protocol/semaphore/blob/main/scripts/download-snark-artifacts.ts

File: README.md [hunk 3]
```diff
@@ -78,6 +78,16 @@ yarn deploy --semaphore <semaphore-address> --group <group-id> --network goerli
 > **Warning**  
 > The group id is a number!
 
+#### Steps to follow every time the smart contract is deployed:
```
Comment (cedoor) [inline README.md:81]:
```suggestion

```
The previous list could continue.

File: README.md [hunk 4]
```diff
@@ -78,6 +78,16 @@ yarn deploy --semaphore <semaphore-address> --group <group-id> --network goerli
...
+
+1. Change the `CONTRACT_ADDRESS` (with the new contract address) and `GROUP_ID` (with the group id) in the `.env` file.
+
+2. Change the `address` (with the new contract address) and `startBlock` (with the block number of the transaction where the contract was created) in the `subgraph.yaml` file.
+
+3. Deploy the subgraph again.
+
+4. Copy the `Feedback.sol` file (contract artifacts) from `apps/contracts/build/contracts/contracts/Feedback.sol` folder to `apps/subgraph/contract-artifacts` and `apps/web-app/contract-artifacts `folders manually or in the root folder, run: `yarn copy:contract-artifacts` to do it automatically.
```
Comment (cedoor) [inline README.md:89]:
```suggestion
3. Change the `address` (with the new contract address) and `startBlock` (with the block number of the transaction where the contract was created) in the `apps/subgraph/subgraph.yaml` file.

4. Copy your contract artifacts from `apps/contracts/build/contracts/contracts` folder to `apps/subgraph/contract-artifacts` and `apps/web-app/contract-artifacts `folders manually. Or run `yarn copy:contract-artifacts` in the project root to do it automatically.

5. Deploy the subgraph again.
```

The subgraph should be deployed only after updating the contract artifacts.

File: README.md [hunk 5]
```diff
@@ -78,6 +78,16 @@ yarn deploy --semaphore <semaphore-address> --group <group-id> --network goerli
 > **Warning**  
 > The group id is a number!
 
+#### Steps to follow every time the smart contract is deployed:
+
+1. Change the `CONTRACT_ADDRESS` (with the new contract address) and `GROUP_ID` (with the group id) in the `.env` file.
+
```
Comment (cedoor) [inline README.md:84]:
```suggestion
```

This should already be explained in the previous point (?).

**Review by cedoor — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (vplasencia) [reply README.md:81]: True

**Review by cedoor — APPROVED** [review]

**Pull request merged**